src/frontend/src/pages/MyGroup/PendingApplications.jsx

```diff
@@ -1,33 +1,93 @@
+    };
+
     return (
         <SmsDataTable rows={applications} headers={headers} className='groups-page__datatable'>
-            {({ rows, headers, getHeaderProps, getTableProps }) => (
-                <TableContainer
-                    title='Pending Group Application'
-                    description='This is a list of all the group application for this semester.'
-                >
-                    <Table {...getTableProps()}>
-                        <SimpleHeader headers={headers} getHeaderProps={getHeaderProps} />
-                        <TableBody>
-                            {rows.map(row => (
-                                <SimpleRow key={row.id} row={row} />
-                            ))}
-                        </TableBody>
-                    </Table>
-                </TableContainer>
-            )}
+            {({
+                rows,
+                headers,
+                getHeaderProps,
+                getTableProps,
+                getSelectionProps,
+                getRowProps,
+                getToolbarProps,
+                selectedRows,
+                getBatchActionProps
+            }) => {
+                const batchActionProps = getBatchActionProps();
+
+                return (
+                    <TableContainer
+                        title='Pending Group Application'
+                        description='This is a list of all the group application for this semester.'
+                    >
+                        <TableToolbar {...getToolbarProps()}>
+                            <TableBatchActions {...batchActionProps}>
+                                <TableBatchAction
+                                    kind='danger'
+                                    tabIndex={batchActionProps.shouldShowBatchActions ? 0 : -1}
+                                    renderIcon={Close}
+                                    onClick={batchActionClick(selectedRows)}
+                                >
+                                    Reject
+                                </TableBatchAction>
+                            </TableBatchActions>
+                        </TableToolbar>
+                        <Table {...getTableProps()}>
+                            <TableHead>
+                                <TableRow>
+                                    <TableSelectAll {...getSelectionProps()} />
+                                    {headers.map((header, i) => (
+                                        <TableHeader key={i} {...getHeaderProps({ header })}>
+                                            {header.header}
+                                        </TableHeader>
+                                    ))}
+                                </TableRow>
+                            </TableHead>
+                            <TableBody>
+                                {rows.map((row, i) => (
+                                    <TableRow key={i} {...getRowProps({ row })}>
```

Comment:
The key can be the `row.id`, which is more explicit than `[0, 1, 2, ..., n]` but `SimpleRow` already does that for us.